src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c

```diff
@@ -27,63 +67,661 @@ const ActorInit Obj_Oshihiki_InitVars = {
+    { { 106, 120, 110 }, { 104, 80, 20 }, { 0, 0, 0 }, { 0, 0, 0 } },               // dodongos cavern
+    { { 142, 99, 86 }, { 72, 118, 96 }, { 0, 0, 0 }, { 0, 0, 0 } },                 // forest temple
+    { { 210, 150, 80 }, { 210, 170, 80 }, { 0, 0, 0 }, { 0, 0, 0 } },               // fire temple
+    { { 102, 144, 182 }, { 176, 167, 100 }, { 100, 167, 100 }, { 117, 97, 96 } },   // water temple
+    { { 232, 210, 176 }, { 232, 210, 176 }, { 232, 210, 176 }, { 232, 210, 176 } }, // spirit temple
+    { { 135, 125, 95 }, { 135, 125, 95 }, { 135, 125, 95 }, { 135, 125, 95 } },     // shadow temple
+    { { 255, 255, 255 }, { 255, 255, 255 }, { 255, 255, 255 }, { 255, 255, 255 } }, // ganons castle
+    { { 232, 210, 176 }, { 232, 210, 176 }, { 232, 210, 176 }, { 232, 210, 176 } }, // gerudo training grounds
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/func_80B9C340.s")
+s16 sScenes[] = {
+    SCENE_YDAN,      // deku
+    SCENE_DDAN,      // dodongos cavern
+    SCENE_BMORI1,    // forest temple
+    SCENE_HIDAN,     // fire
+    SCENE_MIZUSIN,   // water
+    SCENE_JYASINZOU, // spirit
+    SCENE_HAKADAN,   // shadow
+    SCENE_GANON,     // ganons tower
+    SCENE_MEN,       // gerudo training grounds
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/func_80B9C368.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/func_80B9C584.s")
+Vec3f sColCheckPoints[5] = {
+    { 29.99f, 1.01f, -29.99f }, { -29.99f, 1.01f, -29.99f }, { -29.99f, 1.01f, 29.99f },
+    { 29.99f, 1.01f, 29.99f },  { 0.0f, 1.01f, 0.0f },
+}; // The vertices and center of the bottom face
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/func_80B9C5DC.s")
+Vec2f sFaceVtx[] = {
+    { -30.0f, 0.0f },
+    { 30.0f, 0.0f },
+    { -30.0f, 60.0f },
+    { 30.0f, 60.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/ObjOshihiki_Update.s")
+Vec2f sFaceDirection[] = {
+    { 1.0f, 1.0f },
+    { -1.0f, 1.0f },
+    { 1.0f, -1.0f },
+    { -1.0f, -1.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Oshihiki/ObjOshihiki_Draw.s")
+void ObjOshihiki_InitDynapoly(ObjOshihiki* this, GlobalContext* globalCtx, ColHeader* collision,
+                              DynaPolyMoveFlag moveFlag) {
+    s32 pad;
+    ColHeader* colHeader = NULL;
+    s32 pad2;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(collision, &colHeader);
+    this->dyna.dynaPolyId =
+        DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+    if (this->dyna.dynaPolyId == 0x32 /*BGACTOR_MAX*/) {
```

Comment:
The proper define will be used everywhere once it's introduced, so I think there is no need to put a name we haven't decided to use in a comment. (Relevant in all other places there is a comment included with temporary names from other PRs)
```suggestion
    if (this->dyna.dynaPolyId == 50) {
```